### ISSUE TITLE
test for null data types

### ISSUE THREAD

**Issue by delanym** [stoplightio/spectral#1336]
Null data types are not supported in OpenAPIv3. Please check for them.

**Comment by philsturgeon** [comment]
It does. 😆 

```
 154:17    error  oas3-schema            `type` property should be equal to one of the allowed values: array, boolean, integer, number, object, string. Did you mean number?.
```

**Comment by delanym** [comment]
why isn't it seeing that example is null? Openapi-enforcer does.

    {"components":{"schemas":{"4":{"description":"The amount requested.","example":null,"maximum":999999999999,"type":"integer"}}},"info":{"description":"Notty funny.","title":"Naughty nully","version":"1.0.0"},"openapi":"3.0.2","paths":{"/schema":{"get":{"operationId":"schema","responses":{"200":{"content":{"application/json":{}},"description":"Returns the specification."}}}}}}

**Comment by nulltoken** [comment]
@delanym Could you please provide us with a minimal, but _complete_ repro case (document being linted, cli parameters, ...) that puts under th elight the issue you're seeing?

**Comment by nulltoken** [comment]
Took a closer look at this and this specific use case might indeed not be covered by the oas3 schema.

```yaml
openapi: 3.0.2
info:
  description: Notty funny.
  title: Naughty nully
  version: 1.0.0
paths:
  /schema:
    get:
      operationId: schema
      responses:
        '200':
          content:
            application/json: {}
          description: Returns the specification.
components:
  schemas:
    '4':
      description: The amount requested.
      example: null
      maximum: 999999999999
      type: integer
```

```bash
$ npx @stoplight/spectral lint doc.yaml 
npx: installed 161 in 27.742s
OpenAPI 3.x detected

d:/doc.yaml
  1:1  warning  oas3-api-servers               OpenAPI `servers` must be present and non-empty array.
  1:1  warning  openapi-tags                   OpenAPI object should have non-empty `tags` array.
  2:6  warning  info-contact                   Info object should contain `contact` object.
  8:9  warning  operation-description          Operation `description` must be present and non-empty string.
  8:9  warning  operation-tags                 Operation should have non-empty `tags` array.
 17:9  warning  oas3-unused-components-schema  Potentially unused components schema has been detected.

✖ 6 problems (0 errors, 6 warnings, 0 infos, 0 hints)
```

We might need to special case this in the `examples-*` related rules.

@P0lip @philsturgeon Thoughts?

**Comment by philsturgeon** [comment]
Ah the vague title made it hard to know that you meant “check example values against data types” it sounded like you wanted spectral to complain about `type: null` instead. This is solved with the latest example rules that aren’t merged yet.

**Comment by nulltoken** [comment]
>This is solved with the latest example rules that aren’t merged yet.

@delanym That would be #1313, I believe.

Would you want to give it a try, check it out locally, then run `yarn cli lint {here goes the parameters that trigger the weird behavior you're seeing}`.

@P0lip AFAIR, I can't recall #1313 explicitly testing the `null` example(s) value. And I don't see in https://github.com/stoplightio/spectral/blob/66114ef050630e391735cf5690e13a6745b929b7/src/rulesets/oas/functions/oasExample.ts where this might be dealt with. Any idea?

**Comment by P0lip** [comment]
@nulltoken I think we have a truthy check in all JSON path expressions, therefore null won't be caught.
I'd need to change it to undefined.
Perhaps let's get #1313 in and then address this one.
